### PR TITLE
Changes serviceEndpoint to an object in all cases

### PR DESCRIPTION
**Pull request by TelegramSam**
Signed-off-by: Sam Curren <telegramsam@gmail.com>

### CODE REVIEW

**Review by csuwildcat — APPROVED** [review]

**Comment by mccown** [comment]
I like these changes.  While they will cause code changes to current DIDComm implementations, I do like how the uri, accept, and routingKeys will be encapsulated under the serviceEndpoint.  I also think it's important to keep DIDComm compatible with key DID methods.

**Comment by brianorwhatever** [comment]
This makes what is now 1 service endpoint into a list of uris which should arguably be different service endpoints?

File: docs/spec-files/routing.md [hunk 1]
```diff
@@ -110,44 +110,50 @@ TODO: include details about how DIDComm keys are represented/identified in the D
...
+**serviceEndpoint**: MUST contain an ordered list of objects, each represents a DIDComm Service Endpoint URI and it's associated details. The order of the endpoints SHOULD indicate the DID Document owner's preference in receiving messages. Any endpoint MAY be selected by the sender, typically by protocol availability or preference. A message should be delivered to only one of the endpoints specified.
+
+Each object has the following properties:
+
+**uri**: MUST contain an ordered array with one or more URIs for a transport specified in the [transports] section of this spec, or a URI from Alternative Endpoints. It MAY be desirable to constraint endpoints from the [transports] section so that they are used only for the reception of DIDComm messages. This can be particularly helpful in cases where auto-detecting message types is inefficient or undesirable.
```
Comment (TelegramSam) [inline docs/spec-files/routing.md:138]: single URI per block

**Comment by dhh1128** [comment]
+1 from me

**Comment by mccown** [comment]
Discussed / approved in the DIDComm WG on 20220404.

**Pull request merged**